dotnet-coverage-test-data-import-guide.md

```diff
@@ -1,117 +1,161 @@
+
+    * sonar.cs.vscoveragexml.reportsPaths
+        * Path to Visual Studio Code Coverage report. Multiple paths may be comma-delimited, or included via wildcards.
+        * Ensure that you have converted the code coverage report from Binary into XML (see "Convert the Code Coverage Report from Binary into XML" section on our [[Coverage & Test Data] Generate Reports for C#, VB.net](https://community.sonarsource.com/t/coverage-test-data-generate-reports-for-c-vb-net/9871) Community guide)
+    * sonar.vbnet.vscoveragexml.reportsPaths
+    * sonar.cs.dotcover.reportsPaths
+        * Path to dotCover coverage report.
+    * sonar.cs.opencover.reportsPaths
+        * Path to OpenCover coverage report.
+    * sonar.vbnet.dotcover.reportsPaths
+        * Path to dotCover coverage report.
+    * sonar.vbnet.opencover.reportsPaths
+        * Path to OpenCover coverage report.
+
+* Did you specify the correct path to your code coverage file?
+
+    Make sure you understand the regular expression for matching files and directories:
+    |Symbol|Meaning|
+    |---|---|
+    |?|a single character|
+    |*|any number of characters|
+    |**|any number of directories|
+    
+    Here is a resource to understand some other examples: [Pattern matching guide](https://confluence.atlassian.com/fisheye/pattern-matching-guide-960155410.html)
+
+    Examples of Correct Usage:
+    
+    **Right** ✅
+    
+    ✅ Do use glob patterns to match the ultimately created
+    
+    ```-d:"sonar.cs.vscoveragexml.reportsPaths=**/*.coveragexml”``` or
+    
+    ```-d:"sonar.cs.dotcover.reportsPaths=**/*.html"``` or
+    
+    ```-d:"sonar.vbnet.opencover.reportsPaths=**/someNameCodeCoverage.xml"```
+    
+    ✅ Values can be comma separated
+    
+    ```/d:"sonar.cs.vscoveragexml.reportsPaths=coverage/joe*.coveragexml,otherdir/olivier*.coveragexml"```
+    
+    ✅ `-` and `/` are valid flag markers
+    
+    ```/d:"sonar.verbose=true``` or
+    ```-d:"sonar.verbose=true"``` 
+  
+    ✅ Report path is relative to the directory where you run SonarScanner for MSBuild 
+    
+    ```-d:"sonar.cs.vscoveragexml.reportsPaths=coveragexml"```
+    
+    **Wrong** ❌
+    
+    ❌  Don't point at directories but files
+    
+    ```-d:"sonar.cs.vscoveragexml.reportsPaths=coverage/"```
+        
+    ❌ Don't use absolute paths, favor paths relative to the project root directory
+    
+    ```-d:"sonar.cs.vscoveragexml.reportsPaths=C:\users\joe\project\coverage\*.xml"```
+        
+    ❌ Don't point files that are not under the project root directory, always generate coverage under this home directory
+  
+    ```-d:"sonar.cs.vscoveragexml.reportsPaths=..\coverage\*.xml"```
+
+* Do you know how to get debug logs?
+
+    * Add `/d:"sonar.verbose=true"` to the...
+        * SonarScanner.MSBuild.exe `begin` command to get more detailed logs (TODO======>should mention the alternative for Azure Devops integration, where the Scanner for MSBuild is hidden)
```

Comment:
it's 
```
- task: SonarCloudPrepare@1
  inputs:
    SonarCloud: 'sonarcloud'
    organization: 'joetingsanchali'
    scannerMode: 'MSBuild'
    projectKey: 'joetingsanchali_sonar-scanning-someconsoleapp'
    projectName: 'sonar-scanning-someconsoleapp'
    extraProperties: |
      sonar.cs.vstest.reportsPaths=$(Agent.TempDirectory)\**\*.trx
      sonar.cs.vscoveragexml.reportsPaths=$(Agent.TempDirectory)\**\*.coveragexml
      sonar.verbose=true
```

(I took it from [this awesome answer](https://community.sonarsource.com/t/need-help-for-unit-test-code-coverage-for-sonarcloud/32929/5?u=andrei_epure))

Reply:
That Joe guy sure knows something!